sql/core/src/test/resources/sql-tests/results/postgreSQL/date.sql.out

```diff
@@ -864,15 +864,15 @@ SELECT DATE_TRUNC('DECADE', DATE '0004-12-25')
 -- !query schema
 struct<date_trunc(DECADE, CAST(DATE '0004-12-25' AS TIMESTAMP)):timestamp>
 -- !query output
-0000-01-01 00:07:02
+0000-01-01 00:00:00
 
 
 -- !query
 SELECT DATE_TRUNC('DECADE', TO_DATE('0002-12-31 BC', 'yyyy-MM-dd G'))
 -- !query schema
 struct<date_trunc(DECADE, CAST(to_date('0002-12-31 BC', 'yyyy-MM-dd G') AS TIMESTAMP)):timestamp>
 -- !query output
--0010-01-01 00:07:02
+-0010-01-01 00:00:00
```

Comment:
I was going to ask if there are any behavior changes. This is one, but it looks like it was wrong before?

Are there any others that we know of? Given recent discussions I'm extra cautious about introducing behavior changes I suppose.

Reply:
This had been already described in the SQL migration guide. The numbers are different because internally we use Proleptic Gregorian calendar in `date_trunc` but when we collect data to the driver, we convert internal timestamps to `java.sql.Timestamp`. Actually, we switch to Julian calendar for old dates here.

By settings `localSparkSession.conf.set(SQLConf.DATETIME_JAVA8API_ENABLED.key, true)` in `SQLQueryTestSuite.scala`, I just keep the same calendar in all transformations.

Reply:
This is caused by https://github.com/apache/spark/pull/27494/files#diff-432455394ca50800d5de508861984ca5R340 , not the fix itself, right?

Do we have any known behavior changes caused by this fix?

Reply:
Yes, the change in this particular line is caused by `localSparkSession.conf.set(SQLConf.DATETIME_JAVA8API_ENABLED.key, true)`. But I had to set the settings to fix another test failures like in https://github.com/apache/spark/pull/27494#issuecomment-583675754

Reply:
Do you think it's helpful to do https://github.com/apache/spark/pull/27494#discussion_r376988480 first and then we can know what gets changed exactly by this fix?